rust/osp/src/connection.rs

```diff
@@ -15,9 +15,15 @@ use crate::{
 };
 
 /// Sends a command to the unix socket and returns the response
-pub fn send_command<T: AsRef<Path>>(address: T, cmd: ScanCommand) -> Result<Response, Error> {
+pub fn send_command<T: AsRef<Path>>(address: T, r_timeout: Option<Duration>, cmd: ScanCommand) -> Result<Response, Error> {
     let mut socket = UnixStream::connect(address)?;
     let cmd = cmd.try_to_xml()?;
+    // Timeout can not be zero. If not set, it waits for ever
```

Comment:
That comment confuses me. If the timeout can not be zero but it is just set when it is not zero than it could be zero as it is not overridden?